src/libstore/store-api.cc

```diff
@@ -86,18 +86,24 @@ string storePathToHash(const Path & path)
 void checkStoreName(const string & name)
 {
     string validChars = "+-._?=";
+
+    auto baseError = format("The path name '%2%' is invalid: %3%. "
+        "Path names are alphanumeric and can include the symbols %1% "
+        "and must not begin with a period. "
+        "Note: If you cannot rename '%2%', builtins.path { name = ... } "
+        "can be used to give it an alternative name.") % validChars % name;
```

Comment:
That means that you *can* rename it :-) Also, the `builtins.path` advice is only applicable for sources.